src/index/blockfilterindex.cpp

```diff
@@ -335,7 +336,7 @@ bool BlockFilterIndex::CustomRemove(const interfaces::BlockInfo& block)
     if (!m_db->WriteBatch(batch)) return false;
 
     // Update cached header
-    m_last_header = *Assert(ReadFilterHeader(new_tip.height, new_tip.hash));
+    m_last_header = *Assert(ReadFilterHeader(block.height, block.hash));
```

Comment:
In b3e2e1970e46bcfbe19d4d421bb6866ceab40ca0:

This doesn't seem to be correct. The last header is being set to the block we just disconnected. Also, it is in the wrong commit.

(no need to fix it here, I'm cherry-picking this commit into another branch to push it as a separate PR anyway).

Reply:
This can be closed. I pushed it in https://github.com/bitcoin/bitcoin/pull/32694.